src/Symfony/Component/ObjectMapper/ObjectMapper.php

```diff
@@ -92,21 +92,28 @@ public function map(object $source, object|string|null $target = null): object
 
         $this->objectMap[$source] = $mappedTarget;
         $ctorArguments = [];
-        $constructor = $targetRefl->getConstructor();
-        foreach ($constructor?->getParameters() ?? [] as $parameter) {
-            if (!$parameter->isPromoted()) {
-                continue;
-            }
-
+        $targetConstructor = $targetRefl->getConstructor();
+        foreach ($targetConstructor?->getParameters() ?? [] as $parameter) {
             $parameterName = $parameter->getName();
-            $property = $targetRefl->getProperty($parameterName);
 
-            if ($property->isReadOnly() && $property->isInitialized($mappedTarget)) {
-                continue;
+            if ($targetRefl->hasProperty($parameterName)) {
+                $property = $targetRefl->getProperty($parameterName);
+
+                if ($property->isReadOnly() && $property->isInitialized($mappedTarget)) {
+                    continue;
+                }
             }
 
-            // this may be filled later on see storeValue
-            $ctorArguments[$parameterName] = $parameter->isDefaultValueAvailable() ? $parameter->getDefaultValue() : null;
+            try {
+                $ctorArguments[$parameterName] = $this->getRawValue($source, $parameterName);
+            } catch (NoSuchPropertyException) {
+                /*
+                 * When source has no value for the constructor arguments, then fallback to default. Constructor stays
+                 * responsible for delivering a valid object or throwing exceptions, so rely on PHP runtime exceptions
+                 * for further handling.
+                 */
```

Comment:
I don't think that this comment is very useful its quite straightforward.

Reply:
Agreed, removed it. Wrote this while trying to understand the mapper's internals, as the `map` method is quite lengthy.

Reply:
Agreed, removed it. Wrote this while trying to understand the mapper's internals, as the map method is quite lengthy.